cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CruiseControlReconciler.java

```diff
@@ -165,26 +174,36 @@ protected Future<Void> serviceAccount() {
     }
 
     /**
-     * Manages the Cruise Control Config Map with logging and metrics configuration.
+     * Manages the Cruise Control ConfigMap which contains the following:
+     * (1) Cruise Control server configuration
+     * (2) Cruise Control broker capacity configuration
+     * (3) Cruise Control server logging and metrics configuration
      *
-     * @return  Future which completes when the reconciliation is done
+     * @return Future which completes when the reconciliation is done
      */
-    protected Future<Void> metricsAndLoggingConfigMap() {
-        if (cruiseControl != null)  {
+    protected Future<Void> configMap() {
+        if (cruiseControl != null) {
             return MetricsAndLoggingUtils.metricsAndLogging(reconciliation, configMapOperator, cruiseControl.logging(), cruiseControl.metrics())
                     .compose(metricsAndLogging -> {
-                        ConfigMap logAndMetricsConfigMap = cruiseControl.generateMetricsAndLogConfigMap(metricsAndLogging);
+                        ConfigMap configMap = cruiseControl.generateConfigMap(metricsAndLogging);
+
+                        this.serverConfiguration = configMap.getData().get(CruiseControl.SERVER_CONFIG_FILENAME);
+                        this.serverConfigurationHash = Util.hashStub(serverConfiguration);
+                        this.capacityConfiguration = configMap.getData().get(CruiseControl.CAPACITY_CONFIG_FILENAME);
+                        this.capacityConfigurationHash = Util.hashStub(capacityConfiguration);
+                        this.logging = cruiseControl.logging().loggingConfiguration(reconciliation, metricsAndLogging.loggingCm());
+                        this.loggingHash = Util.hashStub(Util.getLoggingDynamicallyUnmodifiableEntries(logging));
```

Comment:
Why do you need to keep the non-hash values?

Reply:
Removed